gadgets/trace_lsm/program.bpf.c

```diff
@@ -0,0 +1,228 @@
+/* Copyright (c) 2024 The Inspektor Gadget authors */
+
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+
+#include <gadget/buffer.h>
+#include <gadget/macros.h>
+#include <gadget/mntns_filter.h>
+
+/* Since LSM hooks may change with different kernel versions, here we only list the common ones.
+ * TODO: Implement best-effort mechanism and add the remaining ones,
+ * this requires an upstream support: https://github.com/cilium/ebpf/discussions/1470
+ */
+#define FOR_EACH_LSM_HOOK(F)        \
+	F(binder_set_context_mgr)   \
+	F(binder_transaction)       \
+	F(binder_transfer_binder)   \
+	F(binder_transfer_file)     \
+	F(ptrace_access_check)      \
+	F(ptrace_traceme)           \
+	F(capget)                   \
+	F(capset)                   \
+	F(capable)                  \
+	F(quotactl)                 \
+	F(quota_on)                 \
+	F(syslog)                   \
+	F(settime)                  \
+	F(vm_enough_memory)         \
+	F(bprm_creds_for_exec)      \
+	F(bprm_creds_from_file)     \
+	F(bprm_check_security)      \
+	F(bprm_committing_creds)    \
+	F(bprm_committed_creds)     \
+	F(fs_context_dup)           \
+	F(fs_context_parse_param)   \
+	F(sb_alloc_security)        \
+	F(sb_delete)                \
+	F(sb_free_security)         \
+	F(sb_free_mnt_opts)         \
+	F(sb_eat_lsm_opts)          \
+	F(sb_mnt_opts_compat)       \
+	F(sb_remount)               \
+	F(sb_kern_mount)            \
+	F(sb_show_options)          \
+	F(sb_statfs)                \
+	F(sb_mount)                 \
+	F(sb_umount)                \
+	F(sb_pivotroot)             \
+	F(sb_set_mnt_opts)          \
+	F(sb_clone_mnt_opts)        \
+	F(move_mount)               \
+	F(dentry_init_security)     \
+	F(dentry_create_files_as)   \
+	F(path_notify)              \
+	F(inode_alloc_security)     \
+	F(inode_free_security)      \
+	F(inode_init_security)      \
+	F(inode_init_security_anon) \
+	F(inode_create)             \
+	F(inode_link)               \
+	F(inode_unlink)             \
+	F(inode_symlink)            \
+	F(inode_mkdir)              \
+	F(inode_rmdir)              \
+	F(inode_mknod)              \
+	F(inode_rename)             \
+	F(inode_readlink)           \
+	F(inode_follow_link)        \
+	F(inode_permission)         \
+	F(inode_setattr)            \
+	F(inode_getattr)            \
+	F(inode_setxattr)           \
+	F(inode_post_setxattr)      \
+	F(inode_getxattr)           \
+	F(inode_listxattr)          \
+	F(inode_removexattr)        \
+	F(inode_need_killpriv)      \
+	F(inode_killpriv)           \
+	F(inode_getsecurity)        \
+	F(inode_setsecurity)        \
+	F(inode_listsecurity)       \
+	F(inode_getsecid)           \
+	F(inode_copy_up)            \
+	F(inode_copy_up_xattr)      \
+	F(kernfs_init_security)     \
+	F(file_permission)          \
+	F(file_alloc_security)      \
+	F(file_free_security)       \
+	F(file_ioctl)               \
+	F(file_ioctl_compat)        \
```

Comment:
With `6.6.13-100.fc38.x86_64`, I get this error:
> Error: starting operator "oci": starting operator "ebpf": creating eBPF collection: program trace_lsm_file_ioctl_compat: attach LSM/LSMMac: file_ioctl_compat LSM hook not supported


After removing it, it works for me:
```diff
diff --git a/gadgets/trace_lsm/gadget.yaml b/gadgets/trace_lsm/gadget.yaml
index aac627a0..cde8c174 100644
--- a/gadgets/trace_lsm/gadget.yaml
+++ b/gadgets/trace_lsm/gadget.yaml
@@ -140,10 +140,6 @@ ebpfParams:
     key: trace-file_ioctl
     defaultValue: "false"
     description: Trace the file_ioctl LSM hook
-  trace_file_ioctl_compat:
-    key: trace-file_ioctl_compat
-    defaultValue: "false"
-    description: Trace the file_ioctl_compat LSM hook
   trace_file_lock:
     key: trace-file_lock
     defaultValue: "false"
diff --git a/gadgets/trace_lsm/program.bpf.c b/gadgets/trace_lsm/program.bpf.c
index 84fef316..771110f5 100644
--- a/gadgets/trace_lsm/program.bpf.c
+++ b/gadgets/trace_lsm/program.bpf.c
@@ -89,7 +89,6 @@
        F(file_alloc_security)      \
        F(file_free_security)       \
        F(file_ioctl)               \
-       F(file_ioctl_compat)        \
        F(mmap_addr)                \
        F(mmap_file)                \
        F(file_mprotect)            \
```